extractor/src/main/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudParsingHelper.java

```diff
@@ -153,7 +153,7 @@ public static String resolveIdWithEmbedPlayer(String url) throws IOException, Re
         String response = NewPipe.getDownloader().get("https://w.soundcloud.com/player/?url="
                 + URLEncoder.encode(url, "UTF-8"), SoundCloud.getLocalization()).responseBody();
         // handle playlists / sets different and get playlist id via uir field in JSON
-        if (url.contains("sets") && !url.endsWith("sets") && !url.endsWith("sets/"))
+        if (url.contains("/sets/") && !url.endsWith("sets") && !url.endsWith("sets/"))
```

Comment:
Actually I think I was wrong above and that this should be changed to `url.contains("/sets/") && !url.endsWith("/sets") && !url.endsWith("/sets/")` so that it only doesn't get soundcloud.com/artist/sets URLs, while e.g. soundcloud.com/artist/track-ending-with-sets still works.

Reply:
> so that it only doesn't get soundcloud.com/artist/sets

Yeah, this seems to be working, but it will still error out on a widget-related issue, similar to that of https://github.com/TeamNewPipe/NewPipeExtractor/issues/412#issuecomment-708876455 in where it lacks data. Otherwise it seems to be detecting it as a channel type request just fine

Reply:
With changes in f803558d1d3c87f9cefcbec2819615721d71d05a, as URLs need to have their tailing slash removed, it is now impossible for a URL to end in sets/. However I am having a decent struggle determining if ending with `sets` should be checked if `/sets/` is our primary check. This has genuinely caused quite a headache. I have left it in there for safety reasons, but if you think that this would be adequate to be removed, then it shall not be a hassle